### ISSUE TITLE
No package-lock.json?

### ISSUE THREAD

**Issue by jcbhmr** [jimmywarting/whatwg-worker#13]
I notice that in https://github.com/jimmywarting/whatwg-worker/commit/9942af7d01831d1b7e4b7386c7544a66315ae462 I missed the main/master branch naming (oops!). That also lead  to me noticing: Why is there no package-lock.json? 🤔 It seems like the current "best practice" is to commit it... Right?

> This file is intended to be committed into source repositories, and serves various purposes:
> 
> 1. Describe a single representation of a dependency tree such that teammates, deployments, and continuous integration are guaranteed to install exactly the same dependencies.
> 
> 2. Provide a facility for users to "time-travel" to previous states of node_modules without having to commit the directory itself.
> 
> 3. Facilitate greater visibility of tree changes through readable source control diffs.
> 
> 4. Optimize the installation process by allowing npm to skip repeated metadata resolutions for previously-installed packages.
> 
> 5. As of npm v7, lockfiles include enough information to gain a complete picture of the package tree, reducing the need to read package.json files, and allowing for significant performance improvements.

https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json

**Comment by jcbhmr** [comment]
wait-a-minute...  am i dumb;  is there no package-lock.json because there's no dependencies? 🤔

**Comment by jcbhmr** [comment]
No, `npm install` still generates a package-lock.json, it's just in the gitignore

![image](https://github.com/jimmywarting/whatwg-worker/assets/61068799/f8b17f7b-12bd-4aaa-9d4e-0f367b33b5ab)
![image](https://github.com/jimmywarting/whatwg-worker/assets/61068799/cae02f32-e4cf-46fe-8654-1a5d6acb32f6)

_That_ is why it's not in my forked copy!

**Comment by jcbhmr** [comment]
This also obviously breaks the `npm ci` since there's no package-lock.json in the test.yml action. Oh no! Those precious seconds with no `cache: npm` to cache the non-existant node_modules!

**Comment by jcbhmr** [comment]
This is also a great time to mention that there's a standard default .gitignore! https://github.com/github/gitignore/blob/main/Node.gitignore

**Comment by jimmywarting** [comment]
yea, i notice it broke the test too. 
it could be fixed by removing the cache and replacing npm ci with npm install, and that is what i have done.

And yep, i do not have any dependencies, though i had at least one dev dependency or something else, but clearly not anything yet...

---

I have just let package lock be included in all my other projects but i tough to my self if i would try it again for old times sake of not having package lock in the source control as I found it bloating pull requests and commit history a lot. 

---

I like it that NodeJS now comes with a built in test runner. and don't have to install tape or anything that looks like 
![image](https://github.com/jimmywarting/whatwg-worker/assets/1148376/fa7f136e-e67a-42e6-8d38-e9e505cda0eb)

Often times try to avoid anything ljharb have built as he imports a hole lot of es3 ponyfills. for just the smallest thing, thinks we still live in 1995
